amazon-kinesis-client/src/test/java/software/amazon/kinesis/coordinator/SchedulerTest.java

```diff
@@ -234,33 +249,29 @@ public final void testCreateOrGetShardConsumer() {
     // TODO: figure out the behavior of the test.
     @Test
     public void testWorkerLoopWithCheckpoint() throws Exception {
-        final String shardId = "shardId-000000000000";
-        final String concurrencyToken = "concurrencyToken";
         final ExtendedSequenceNumber firstSequenceNumber = ExtendedSequenceNumber.TRIM_HORIZON;
         final ExtendedSequenceNumber secondSequenceNumber = new ExtendedSequenceNumber("1000");
         final ExtendedSequenceNumber finalSequenceNumber = new ExtendedSequenceNumber("2000");
-
-        final List<ShardInfo> initialShardInfo = Collections.singletonList(
-                new ShardInfo(shardId, concurrencyToken, null, firstSequenceNumber));
-        final List<ShardInfo> firstShardInfo = Collections.singletonList(
-                new ShardInfo(shardId, concurrencyToken, null, secondSequenceNumber));
-        final List<ShardInfo> secondShardInfo = Collections.singletonList(
-                new ShardInfo(shardId, concurrencyToken, null, finalSequenceNumber));
-
         final Checkpoint firstCheckpoint = new Checkpoint(firstSequenceNumber, null, null);
+        when(checkpoint.getCheckpointObject(eq(TEST_SHARD_ID))).thenReturn(firstCheckpoint);
 
-        when(leaseCoordinator.getCurrentAssignments()).thenReturn(initialShardInfo, firstShardInfo, secondShardInfo);
-        when(checkpoint.getCheckpointObject(eq(shardId))).thenReturn(firstCheckpoint);
+        when(leaseCoordinator.getAssignments()).thenReturn(
+                Stream.of(firstSequenceNumber, secondSequenceNumber, finalSequenceNumber)
+                        .map(SchedulerTest::constructLease)
+                        .collect(Collectors.toList()));
```

Comment:
```suggestion
        when(leaseCoordinator.getAssignments()).thenReturn(
            ImmutableList.of(
                constructLease(firstSequenceNumber),
                constructLease(secondSequenceNumber),
                constructLease(finalSequenceNumber)));
```
Avoid control flow in unit tests.